src/Searching.js

```diff
@@ -107,6 +107,26 @@ async function localSearch(searchTerm, roomId = undefined) {
     const result = MatrixClientPeg.get()._processRoomEventsSearch(
         emptyResult, response);
 
+    // Restore our encryption info so we can properly re-verify the events.
+    for (let i = 0; i < result.results.length; i++) {
+        const timeline = result.results[i].context.getTimeline();
+
+        for (let j = 0; j < timeline.length; j++) {
+            const ev = timeline[j];
+            if (ev.event.curve25519Key) {
```

Comment:
I assume you have to use this as a test instead of say `isEncrypted()` because it's already marked as encrypted by event type, but just the keys aren't in place yet...?

Reply:
`isEncrypted()` uses the event `type` field to check if the event is encrypted, we're storing the clear event in Seshat and it'll never have the `m.room.encrypted` type.